themes/bootstrap3/templates/RecordDriver/DefaultRecord/cover.phtml

```diff
@@ -1,16 +1,15 @@
-<?php $alt = $this->link ? trim($this->driver->tryMethod('getTitle') ?? '') : ''; ?>
-<?php if ($this->link && !empty($alt)): ?><a href="<?=$this->escapeHtmlAttr($this->link)?>" class="record-cover-link"><?php endif; ?>
+<?php if ($this->link): ?><a href="<?=$this->escapeHtmlAttr($this->link)?>" class="record-cover-link" <?php if (!$this->linkUnique): ?>tabindex="-1"<?php endif; ?>><?php endif; ?>
 <?php /* Display thumbnail if appropriate: */ ?>
 <?php if ($cover): ?>
-  <img src="<?=$this->escapeHtmlAttr($cover); ?>" <?php if ($linkPreview): ?>data-linkpreview="true" <?php endif; ?>class="recordcover" alt="<?=$this->escapeHtmlAttr($alt); ?>">
+  <img src="<?=$this->escapeHtmlAttr($cover); ?>" <?php if ($linkPreview): ?>data-linkpreview="true" <?php endif; ?>class="recordcover" alt="<?=$this->transEscAttr('Cover Image') ?>">
```

Comment:
Might it still make sense to have an empty alt in situations where the image is not linked and/or the link is not unique, to reduce noise, since the cover images are essentially just decoration in these situations?

Reply:
Every link needs to have an accessible name, so if there is a link there also needs to be an alt-text. But I can adapt the code for the case if there is no link.

Reply:
That's fair, but I wonder if the best thing to do would be to put `aria-hidden="true"` on the link for non-unique cover links. This way, our markup can be valid, but we can also hide useless repetitive "Cover Image" announcements from users who won't find them useful. What do you think?

Reply:
My head is spinning a bit, so just to be sure:

- if no link: empty `alt`
- if non-unique link: non-empty `alt` and `aria-hidden`
- if unique link: non-empty `alt` and no `aria-hidden`

That's what I implemented.

I think it is a little weird that the "no link" case is a little less hidden than the "non-unique link" case. An alternative approach could be to always have a non-empty `alt` and add a `<div aria-hidden="true">` instead of the `<a>` for the "no link" case. What do you think?

Reply:
done

Reply:
One last question: since `div` is a block-level element and `a` is not, would it be better to use a `span` here for consistency of layout/flow?

Reply:
That depends on the CSS. I assumed that since the `a` contains an `img` that there is some kind of special styling. In the EDS template it also contains `<div class="ajaxcover">`. But I don't know enough about the context to give a definitive answer. Do you think I should change it?

Note that `a` is no longer an inline element in HTML 5 https://stackoverflow.com/a/1828032.

Reply:
I'll defer to @crhallberg on this question, but it sounds like I'm probably just behind the times in my knowledge of the HTML standards. Thanks for clarifying that!